src/GraphQL/loaders/parseClassMutations.js

```diff
@@ -78,21 +82,36 @@ const load = function(
       args: {
         fields: createFields,
       },
-      type: new GraphQLNonNull(defaultGraphQLTypes.CREATE_RESULT),
-      async resolve(_source, args, context) {
+      type: new GraphQLNonNull(classGraphQLOutputType),
+      async resolve(_source, args, context, mutationInfo) {
         try {
           const { fields } = args;
           const { config, auth, info } = context;
 
           transformTypes('create', fields);
-
-          return await objectsMutations.createObject(
+          const { objectId } = await objectsMutations.createObject(
             className,
             fields,
             config,
             auth,
             info
           );
+          const selectedFields = getFieldNames(mutationInfo);
+          const { keys, include } = parseClassTypes.extractKeysAndInclude(
+            selectedFields
+          );
+
+          return await objectsQueries.getObject(
```

Comment:
My suggestion here is:
1 - Create a new object containing the objectId, createdAt, updatedAt (that are returned by default from the rest create operation) + possibly other values returned from the rest create operation (sometimes it returns more things - when something is added in a trigger for example) + the input fields
2 - Check the selectedFields to compare if the request is asking more than the fields in the object 1
3 - Only do the query if necessary

Thoughts?

Reply:
Yes seems good for me too